web/src/components/PixiMap.tsx

```diff
@@ -147,38 +171,190 @@ const RequiresInitLayer = <T,>({
 
 export const PixiMap = <T,>({
   height = "100%",
-  defaultCenter = DEFAULT_CENTER,
-  defaultZoom = DEFAULT_ZOOM,
+  selectionDropdownLeft = "10px",
+  selectionDropdownTop = "10px",
+  showCitySelectionDropDown = true,
+  defaultCenter,
   useRenderer,
   options,
-  ...rest
 }: PixiMapProps<T>) => {
-  const [center, setCenter] = useState(defaultCenter);
-  const [zoom, setZoom] = useState(defaultZoom);
+  const [center, setCenter] = React.useState(defaultCenter || DEFAULT_CENTER);
+  const [zoom] = React.useState(DEFAULT_ZOOM);
+  const { colorMode } = useColorMode();
+  const [lastSelectedCityId, setLastSelectedCityId] =
+    useRecoilState(selectedCity);
+  const [selectedCities] = useRecoilState(selectedCitiesFromRecoil);
+  const [isUpdating] = useRecoilState(isUpdatingCities);
+  const [forceUpdateMap, setForceUpdateMap] = React.useState(false);
+  const { connected } = useConnectionState();
+  const [dropdownDisabledMsg, setDropdownDisabledMsg] = React.useState("");
+
+  React.useEffect(() => {
+    if (!connected) {
+      setDropdownDisabledMsg("Please configure connection to change map city");
+    } else if (selectedCities.length <= 0) {
+      setDropdownDisabledMsg(
+        "Select atleast one city from dashboard location section"
+      );
+    } else if (isUpdating) {
+      setDropdownDisabledMsg("City list is updating please wait");
+    }
+  }, [connected, isUpdating, selectedCities]);
+
+  const [lastSelectedCityDetails, setLastSelectedCityDetails] = React.useState(
+    undefined as City | undefined
+  );
+
+  let selectionValue = { label: "Select city", value: { id: -1 } };
+  if (lastSelectedCityDetails) {
+    selectionValue = {
+      label: lastSelectedCityDetails.name,
+      value: lastSelectedCityDetails,
+    };
+  }
+
+  let centerValue: [number, number] | undefined = undefined;
+  if (lastSelectedCityDetails && !defaultCenter) {
+    centerValue = [
+      lastSelectedCityDetails.centerLat,
+      lastSelectedCityDetails.centerLon,
+    ];
+  } else if (defaultCenter) {
+    centerValue = [defaultCenter[1], defaultCenter[0]];
+  }
+
+  React.useEffect(() => {
+    setForceUpdateMap(true);
+    setLastSelectedCityDetails(
+      selectedCities.find((c) => c.id === lastSelectedCityId)
+    );
+  }, [selectedCities, lastSelectedCityId]);
+
+  React.useEffect(() => {
+    setForceUpdateMap(false);
+    if (lastSelectedCityDetails && !defaultCenter) {
+      setCenter([
+        lastSelectedCityDetails.centerLat,
+        lastSelectedCityDetails.centerLon,
+      ]);
+    }
+  }, [defaultCenter, lastSelectedCityDetails]);
 
   return (
-    <Box borderRadius="lg" overflow="hidden" height={height} {...rest}>
-      <Map
-        dprs={[1, 2]}
-        provider={stamenProvider("toner-lite")}
-        attribution={stamenAttribution}
-        maxZoom={20}
-        center={center}
-        zoom={zoom}
-        onBoundsChanged={({ center, zoom }) => {
-          setCenter(center);
-          setZoom(zoom);
-        }}
```

Comment:
Why did you delete this? Is it no longer necessary? If so, there is no point in using `React.useState` if we are never changing the value of `zoom`, or am I missing something?

Reply:
made some changes and this is back

Reply:
made some changes and this is back